std/variant.d

```diff
@@ -74,28 +74,48 @@ import std.meta, std.traits, std.typecons;
 
 /++
     Gives the `sizeof` the largest type given.
+
+    See_Also: https://forum.dlang.org/thread/wbpnncxepehgcswhuazl@forum.dlang.org?page=1
   +/
-template maxSize(T...)
+template maxSize(Ts...)
 {
-    static if (T.length == 1)
-    {
-        enum size_t maxSize = T[0].sizeof;
-    }
-    else
-    {
-        import std.algorithm.comparison : max;
-        enum size_t maxSize = max(T[0].sizeof, maxSize!(T[1 .. $]));
+    align(1) union Impl {
+        static foreach (i, T; Ts) {
+            static if (!is(T == void))
+                mixin("T _field_" ~ i.stringof ~ ";");
```

Comment:
mixin can implictly concatenate and convert i : `mixin("T _field_", i, ";");`

Reply:
Thanks